src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -837,6 +839,19 @@ public function fromDateTime($value)
         );
     }
 
+    /**
+     * Return a timestamp as DateTime object with the current date.
+     *
+     * @param  mixed  $value
+     * @return false|\Illuminate\Support\Carbon
+     */
+    protected function asTime($value)
+    {
+        return $value instanceof CarbonInterface
+            ? Date::instance($value)
+            : Date::createFromFormat('H:i:s', $value)->startOfSecond();
```

Comment:
I added a microsecond reset because without it the time comparison would be wrong.

The TIME field type of the database does not store information about milliseconds, and when the `now ()` functions are called in sequence, the values will be different.
The `startOfSecond ()` method leads them to a single value, as a result of which the comparison will be correct.

The result of the verification showed here: https://github.com/laravel/framework/pull/30931#pullrequestreview-336364360